webxr: create glwindow with Rc window and without rendering context

wusyong opened the pull request:
<!-- Please describe your changes on the following line: -->
Companion PR of https://github.com/servo/webxr/pull/257
Part of https://github.com/servo/servo/issues/34522

This PR clarifies rendering context usage in Weber. WebXR's window creates its own Surfman context, so it doesn't really need to pass the rendering context's information. This should simplify the usage for rendering context.

---
<!-- Thank you for contributing to Servo! Please replace each `[ ]` by `[X]` when the step is complete, and replace `___` with appropriate data: -->
- [x] `./mach build -d` does not report any errors
- [x] `./mach test-tidy` does not report any errors

<!-- Either: -->
- [x] These changes do not require tests because but servoshell should be built successfully.

<!-- Also, please make sure that "Allow edits from maintainers" checkbox is checked, so that we can help you if you get stuck somewhere along the way.-->

<!-- Pull requests that do not address these steps are welcome, but they will require additional verification as part of the review process. -->

wusyong committed:
Create webxr glwindow with Rc window

Signed-off-by: Wu Yu Wei <yuweiwu@pm.me>

wusyong commented on Cargo.toml:
```diff
@@ -222,5 +222,6 @@ codegen-units = 1
 #
-# [patch."https://github.com/servo/<repository>"]
-# <crate> = { path = "/path/to/local/checkout" }
+[patch."https://github.com/servo/webxr"]
```
This will be removed once PR in webxr is merged.

jdm commented on ports/servoshell/desktop/events_loop.rs:
```diff
@@ -189,23 +189,3 @@ impl Drop for EventLoopGuard {
     }
 }
-
-// Helper function to safely use the current event loop
```
Can we also remove the CURRENT_EVENT_LOOP theead-local and EventLoopGuard type?

wusyong replied:
Ah right. The guard type is obsolete after this change.

jdm reviewed (approved):
This is a nice simplification. Which platforms have you tested the glwindow backend on after these changes?

wusyong committed:
Remove obselte gurad type

Signed-off-by: Wu Yu Wei <yuweiwu@pm.me>

wusyong commented:
@jdm Hmm okay I noticed I should hide the window at first. I was testing on wayland and it won't show the window if it's empty, and I thought it worked.
I tried to test some webxr demo sites from servo wiki but they won't work. But in the meantime, the main branch didn't work either (I've make sure it used glwindow instead of openxr port).
Is there any instruction I could follow to test glwindow?

wusyong committed:
Update GlWindow trait method

Signed-off-by: Wu Yuwei <yuweiwu@pm.me>

wusyong committed:
Update how webxr discorvery is created

Now glwindow will create a hidden window. It's better to not use it
unless we really want to use this port.

Signed-off-by: Wu Yu Wei <yuweiwu@pm.me>

wusyong commented:
Here's how it works on wayland to prove the connection is created from provided display handle (since Window's connetion is an empty type)

![Screenshot from 2025-01-02 18-53-02](https://github.com/user-attachments/assets/3540be6a-a092-4c62-8e6c-de1d4346c422)

wusyong committed:
Link back to upstream webxr repo

Signed-off-by: Wu Yu Wei <yuweiwu@pm.me>
